evm/utils/db.py

```diff
@@ -8,50 +8,62 @@
     EMPTY_SHA3,
 )
 
+from evm.rlp.blocks import BaseBlock
+from evm.rlp.headers import BlockHeader
 
-def make_block_number_to_hash_lookup_key(block_number):
+from typing import Any
+
+# from evm.db.chain import BaseChainDB
```

Comment:
@gsalgado 

I was fighting really hard about circular dependencies with python here. Maybe you know a way out?
So the problem is that `evm.db.chain` imports this file (`evm.utils.db.py`) hence importing `BaseChainDB` here would create a circular dependency. 

Here's what I tried but failed:

- use forward references https://www.python.org/dev/peps/pep-0484/#forward-references
- use a `NewType` pointing to forward reference
- import just `evm` and then type it as `evm.db.chain.BaseChainDB` where it is needed.

Do you have any more ideas on how I could resolve this?

Reply:
http://mypy.readthedocs.io/en/latest/common_issues.html#import-cycles

Reply:
:heart: that should work! Thank you